src/Microsoft.DotNet.Build.Tasks/GenFacadesTask.cs

```diff
@@ -0,0 +1,83 @@
+﻿using System;
+using Microsoft.Build.Utilities;
+using Microsoft.Build.Framework;
+using GenFacades;
+using Microsoft.Cci.Extensions;
+using System.Linq;
+
+namespace Microsoft.DotNet.Build.Tasks
+{
+    /// <summary>
+    /// Runs GenFacades In-Proc.
+    /// </summary>
+    public sealed class GenFacadesTask : Task
+    {
+        [Required]
+        public string Seeds { get; set; }
+
+        [Required]
+        public string Contracts { get; set; }
+
+        [Required]
+        public string FacadePath { get; set; }
+
+        public Version AssemblyFileVersion { get; set; }
+
+        public bool ClearBuildAndRevision { get; set; }
+
+        public bool IgnoreMissingTypes { get; set; }
+
+        public bool BuildDesignTimeFacades { get; set; }
+
+        public string InclusionContracts { get; set; }
+
+        public string SeedLoadErrorTreatment { get; set; }
+
+        public string ContractLoadErrorTreatment { get; set; }
+
+        public ITaskItem[] SeedTypePreferencesUnsplit { get; set; }
+
+        public bool ForceZeroVersionSeeds { get; set; }
+
+        public bool ProducePdb { get; set; } = true;
+
+        public string PartialFacadeAssemblyPath { get; set; }
+
+        public override bool Execute()
+        {
+            try
+            {
+                ErrorTreatment seedLoadErrorTreatment = ErrorTreatment.Default;
+                ErrorTreatment contractLoadErrorTreatment = ErrorTreatment.Default;
+
+                if (SeedLoadErrorTreatment != null)
+                {
+                    seedLoadErrorTreatment = (ErrorTreatment)Enum.Parse(typeof(ErrorTreatment), SeedLoadErrorTreatment);
+                }
+                if (ContractLoadErrorTreatment != null)
+                {
+                    contractLoadErrorTreatment = (ErrorTreatment)Enum.Parse(typeof(ErrorTreatment), ContractLoadErrorTreatment);
+                }
+
+                string[] seedTypePreferencesUnsplit = null;
+                if (SeedTypePreferencesUnsplit != null)
+                {
+                    seedTypePreferencesUnsplit = SeedTypePreferencesUnsplit.Select(iti => iti.ItemSpec).ToArray();
+                }
+
+                Generator.GenerateFacades(
```

Comment:
How are we detecting if errors occur? We should make sure the exceptions do not flow out because that will derail our builds but instead just have some return code that indicated success or failure.